exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetBooleanPredicate.java

```diff
@@ -40,51 +40,72 @@ private ParquetBooleanPredicate(String name, List<LogicalExpression> args, Expre
   }
 
   @SuppressWarnings("unchecked")
-  private static <C extends Comparable<C>> LogicalExpression createAndPredicate(
-      String name,
-      List<LogicalExpression> args,
-      ExpressionPosition pos
-  ) {
+  private static <C extends Comparable<C>> LogicalExpression createAndPredicate(String name, List<LogicalExpression> args, ExpressionPosition pos) {
     return new ParquetBooleanPredicate<C>(name, args, pos) {
+      /**
+       * Evaluates a compound "AND" filter on the statistics of a RowGroup (the filter reads "filterA and filterB").
+       * Return value :<ul>
+       *   <li>ALL : only if all filters return ALL
+       *   <li>NONE : if one filter at least returns NONE
+       *   <li>SOME : all other cases
+       * </ul>
+       */
       @Override
-      public boolean canDrop(RangeExprEvaluator<C> evaluator) {
-        // "and" : as long as one branch is OK to drop, we can drop it.
+      public RowsMatch matches(RangeExprEvaluator<C> evaluator) {
+        RowsMatch childMatch, lastMatch = RowsMatch.NONE;
```

Comment:
Avoid declaring `childMatch` and `lastMatch` on the same line. Move `childMatch` declaration to the assignment (line 58). Consider renaming `lastMatch` to `andResultMatch`.

Reply:
Done